### PR TITLE
WIP: Use all channels in fnirs conversion. Don't exclude bads

### PR DESCRIPTION
**Pull request by rob-luke**
#### What does this implement/fix?
FNIRS channels that had been marked as bad were not being converted. I think all channels should be converted to the final hbo/hbr, then the downstream processing can choose what to do with bad channels.

Adds two internal function for checking that nirs channel labelling of bads is consistent. And nother function that spreads bad marking to ensure it is consistent.

### CODE REVIEW

**Comment by rob-luke** [comment]
@larsoner could you please review. These errors don’t seem to be related to the code changes. Thanks

File: mne/preprocessing/nirs/nirs.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def _check_channels_ordered(raw, freqs):
     """Check channels followed expected fNIRS format."""
     # Every second channel should be same SD pair
     # and have the specified light frequencies.
-    picks = _picks_to_idx(raw.info, 'fnirs_od')
+    picks = _picks_to_idx(raw.info, 'fnirs_od', exclude=[])
```
Comment (larsoner) [inline mne/preprocessing/nirs/nirs.py]: At some stage of processing, do we need to do something about pairs of channels where only one is marked bad? Basically whatever processing step(s) actually combine the data from two channels should probably "spread the badness", i.e., if one of them is marked bad before the operation, the operation can do its thing, but both should be marked as bad afterward.

Comment (rob-luke) [reply mne/preprocessing/nirs/nirs.py]: That's a great idea. I will write two function. One that checks that the labeling of bads is consistent across pairs of optodes, something like `_fnirs_check_bads`. And another function that spreads bads if only a single optode frequency was marked bad, something like `_fnirs_spread_bads`. What do you think of this? I was thinking to not expose these functions as they seem advanced and the automated quality check functions like `scalp_coupling_index` should behave correctly and also mark the correct number of bads per optode.

Comment (rob-luke) [reply mne/preprocessing/nirs/nirs.py]: Can I do add those functions in another PR? This one seems like a bug fix, the new functions are new features.

Comment (rob-luke) [reply mne/preprocessing/nirs/nirs.py]: > Can I do add those functions in another PR? This one seems like a bug fix, the new functions are new features.

Ignore, I just did it here.

**Comment by rob-luke** [comment]
@larsoner This is now ready for review. Thanks

**Comment by larsoner** [comment]
Looks reasonable to me, thanks @rob-luke !

**Comment by rob-luke** [comment]
#7057

**Pull request merged**